### PR TITLE
Adding support for Node 4.2.x LTS

### PR DESCRIPTION
**Pull request by edsadr**
Ref #5821
- Adding Node 4.2 in supported engines at package.json
- Adding Node 4.2 version in Travis

### CODE REVIEW

**Comment by edsadr** [comment]
@ErisDS : this is the basic PR adding Node v4.2 to supported engines and also including this in Travis, I want to discuss here the other things you want to add and another suggested:

*  ```PR adding a pre-install hook to handle node version support, instead of our startup check, to make it easier for people to understand why Ghost isn't installing before they hit a failing dep like node-sqlite3```
We can get this done running in the preinstall hook but via a Node script instead of Bash, I would basically get the current process version and would compare against the min and max version supported, would this work?

* ```Allowing users to run Ghost with unsupported Node versions with a warning when installing and another at the admin panel (same as the current email system warning)```
Basically this would be doing changing the nodeVersion protection in the startup-check.js making it check just the min supported version...

**Comment by ErisDS** [comment]
This PR doesn't do what was asked for:

> I'd love to see a PR adding node 4.2.0 to the travis build - this is more than a simple text change as the startup checks will need to be overridden in some way when in test mode.
